### ISSUE TITLE
does not merge correctly using merge_with_ir

### ISSUE THREAD

**Issue by shuchaa** [scverse/scirpy#364]
### Description of the bug

merging single cell data with TCR data using ir.pp.merge_with_ir(adata, adata_tcr) does not merge correctly as I get two different samples per obs.names, instead of it being the same sample.

### Minimal reproducible example

```python
ir.pp.merge_with_ir(adata, adata_tcr) #try with different samples that were concatenated separately in adata and adata_tcr
```

### The error message produced by the code above

```pytb
sample_x		sample_y
obs_names																					
AAAC-----AGTC-1	sample1	sample2
```

### Version information

```txt
anndata     0.8.0
scanpy      1.9.1
numpy                       1.22.4
pandas                      1.4.3
scipy                       1.9.0
scirpy                      0.11.1
Python 3.9.13 
Linux-5.4.0-1081-aws-x86_64-with-glibc2.27
```

**Comment by grst** [comment]
Hi @shuchaa,

by default the function only joins the two objects on the `obs_names`. You can join on additional columns by specifying the `on` parameter, i.e. in your case

```
ir.pp.merge_with_ir(adata, adata_tcr, on=["sample"])
```

LMK if this works for you!

Best,
Gregor

**Comment by grst** [comment]
With the new data structure (currently available as release candidate in v0.13.0rc1), we are moving away from merging the objects and use [MuData](https://muon.readthedocs.io/en/latest/notebooks/quickstart_mudata.html) instead. 

See https://scverse.org/scirpy/tags/v0.13.0rc1/data-structure.html#working-with-multimodal-data. 

I'm therefore closing this issue.
